src/main/java/io/vertx/core/http/impl/Http2ServerResponseImpl.java

```diff
@@ -709,4 +712,15 @@ public HttpServerResponse push(HttpMethod method, String host, String path, Mult
   public HttpServerResponse push(HttpMethod method, String path, Handler<AsyncResult<HttpServerResponse>> handler) {
     return push(method, host, path, handler);
   }
+
+  @Override
+  public HttpServerResponse setStreamPriority(StreamPriority streamPriority) {
+    if(!streamPriority.equals(stream.getStreamPriority())) {
+      stream.setStreamPriority(streamPriority);
+      if(headWritten) {
```

Comment:
I think it would be better to let the stream figure out or not whether it needs to send a priority frame internally. Indeed the HTTP/2 stream knows if the first header frame has been sent or not

Reply:
There is no variable in VertxHttp2Stream. I could not see the way to figure out if the head was written without adding state variable, so I though it will be simpler to reuse Http2ServerResponseImpl.headWritten.